### DISCUSSION TITLE
Get affected rows for delete() or update()

### DISCUSSION THREAD

**Discussion by maxgalbu** [adonisjs/core#3196] [Help] [answered]
```
const affectedRows = await User.query()
      .where("type", sometype)
      .delete();
const affectedRows2 = await User.query()
      .where("type", type)
      .update("name", "hello world");
```

Is it possible? Was possible in v4

**Comment by ConstaXI** [comment]
Not sure if this is what you want, but affectedRows2 is just the number of affected rows by update, to get the row itself you need to provide a returning argument, like that:

`const affectedRows2 = await User.query().where("type", type).update("name", "hello world", "*");`

Unfortunately, the delete method does not have this option, but you can also query users and then delete, so you'll know who will be deleted

**Comment by thetutlage** [comment, marked as answer]
If the database supports returning columns, then you can use the `returning` method with both the both and the delete methods.

```ts
await User.query().where("type", type).returning(['id', 'name']).update()
await User.query().where("type", type).returning(['id', 'name']).del()
```

**Reply by maxgalbu** [reply]
I have an error when using returning:

```
Property 'returning' does not exist on type 'ModelQueryBuilderContract<typeof Users, Users>'.
```

**Reply by thetutlage** [reply]
Yeah, that was missing in the types. Just upgrade to the latest of Lucid, I have fixed it

**Reply by maxgalbu** [reply]
Thanks for the quick fix. 
So, looking at the code, returning() is not supported by neither mysql nor sqlite?

**Reply by thetutlage** [reply]
Yup, the underlying database has to support it. Otherwise there is no way to get what was updated. You can check with a direct query against your database

**Reply by maxgalbu** [reply]
@thetutlage I've accepted your answer but actually, I just want the _number_ of affected rows which is something that by default mysql returns. And it seems that also knex does it, at least for mysql: 

https://github.com/knex/knex/blob/57395af6eb28b5eb4fc780fe83188e52c0fb4fb7/lib/dialects/mysql/index.js#L158

The return value of update/del is ignored in adonisjs/lucid:

delete: https://github.com/adonisjs/lucid/blob/9b35e1399790d7d20dab5c49ae1cd0e1a170f7c8/src/Database/QueryBuilder/Database.ts#L114
update: https://github.com/adonisjs/lucid/blob/9b35e1399790d7d20dab5c49ae1cd0e1a170f7c8/src/Database/QueryBuilder/Database.ts#L188

Returning `this` allows to concatenate commands after del() or update() ... is this wanted?

**Reply by thetutlage** [reply]
I just gave it a test run and seems to be working fine as follows.

```ts
  const [affectedRows] = await User.query().where('id', 3).update({ email: 'virk@adonisjs.com' })
  console.log(affectedRows)
```
